flink-runtime/src/main/java/org/apache/flink/runtime/zookeeper/ZooKeeperStateHandleStore.java

```diff
@@ -474,13 +515,22 @@ public void releaseAndTryRemoveAll() throws Exception {
     @Override
     public void release(String pathInZooKeeper) throws Exception {
         final String path = normalizePath(pathInZooKeeper);
+        final String lockPath = getLockPath(path);
         try {
-            client.delete().forPath(getLockPath(path));
-        } catch (KeeperException.NoNodeException ignored) {
-            // we have never locked this node
+            deleteIdempotently(lockPath);
         } catch (Exception e) {
-            throw new Exception(
-                    "Could not release the lock: " + getLockPath(pathInZooKeeper) + '.', e);
+            throw new Exception("Could not release the lock: " + lockPath + '.', e);
+        }
+    }
+
+    private void deleteIdempotently(String path) throws Exception {
```

Comment:
It took me a while to understand what's meant as idempotent here. How about:

```suggestion
    private void deleteIfExists(String path) throws Exception {
```

?

Reply:
I like that. It's probably more descriptive than the `deleteIdempotently` method name 👍